src/Adapter/Product/CommandHandler/BulkUpdateProductStatusHandler.php

```diff
@@ -66,7 +76,28 @@ public function handle(BulkUpdateProductStatusCommand $command): void
      */
     protected function handleSingleAction(ProductId $productId, $command = null)
     {
-        $this->productStatusUpdater->updateStatus($productId, $command->getNewStatus());
+        $product = $this->productMultiShopRepository->getByShopConstraint($productId, $command->getShopConstraint());
+        $wasVisibleOnSearch = $this->productIndexationUpdater->isVisibleOnSearch($product);
+        $wasActive = (bool) $product->active;
+
+        $product->active = $command->getNewStatus();
+        $this->productMultiShopRepository->partialUpdate(
+            $product,
+            ['active'],
+            $command->getShopConstraint(),
+            CannotUpdateProductException::FAILED_UPDATE_STATUS
+        );
+
+        // Reindexing is costly operation, so we check if properties impacting indexation have changed and then reindex if needed.
+        if (
+            $wasVisibleOnSearch !== $this->productIndexationUpdater->isVisibleOnSearch($product)
+            || $wasActive !== (bool) $product->active
```

Comment:
following the above comment this status check would also be not needed anymore, because we will only reach this part once the status is changed.

Reply:
See answer above, but I'm not sure the optimization is worth the risk for the partial update it's a really small update on one field. And for the indexation, which is the one that is expensive, it's already handled